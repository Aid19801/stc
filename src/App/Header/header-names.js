export const headerNames = {
    sections: [
        { name: 'home', link: '#' },
        { name: 'about', link: '#' },
        { name: 'media', link: '#' },
        { name: 'gallery', link: '#' },
        { name: 'contact', link: '#' }
    ]
}