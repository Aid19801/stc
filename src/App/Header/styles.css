
.container {
    border: 1px solid black;
    padding-top: 1vh;
    padding-bottom: 1vh;
    background-color: #804715;

    display: flex;
    flex-flow: row wrap;
    justify-content: flex-end;
    align-items: center; /* centre of vertical/container */

    opacity: 1;
    animation-name: fadein;
    animation-duration: 0.5s;
}


.navbar-item {
    flex: 1;
    flex-grow: 1;
    margin: auto;
    color: white;
    text-shadow: 1px 1px 1px black;
    font-size: 3vh;
    text-align: center;
    margin-top: 0;

    animation-name: bouncein;
    animation-duration: 0.4s;
}

.resp-nav {
    text-align: center;
    padding-left: 0;

    animation-name: fadein;
    animation-duration: 2s;
}

.resp-navbar-item {
    flex: 1;
    flex-grow: 1;
    color: white;
    text-shadow: 1px 1px 1px black;
    font-size: 3vh;
    text-align: center;
    border: 3px solid black;
    margin-right: 10px;
    margin-left: 10px;
    margin-top: 5px;
    margin-bottom: 5px;
    list-style-type: none;
}

.resp-navbar-item:hover {
    background-color: orange;
}

.hamburger {
    color: white;
    text-shadow: 1px 1px 1px black;
    font-size: 3vh;
    margin-top: 3vh;
    margin-bottom: 3vh;

    display: none;
}



@keyframes fadein {
    from {
        opacity: 0;
    }
    to {
        opacity: 1;
    }
}

@keyframes bouncein {
    from {
        margin-top: -7vh;
    }
    to {
        margin-top: 0;
    }
}

@media only screen and (max-width: 800px) {
    .container {
        justify-content: space-around;
    }
}


@media only screen and (min-width: 501px) {
    .resp-navbar-item {
        display: none;
    }
}

@media only screen and (max-width: 500px) {
    .container {
        flex-direction: column; 
        display: block;

        padding-top: 0.2vh;
        padding-bottom: 0.2vh;
    }

    .navbar-item {
        display: none;
    }

    .hamburger {
        display: flex;
    }
}