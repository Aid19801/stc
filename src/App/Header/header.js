import React, { Component } from 'react';
import { headerNames } from './header-names';
import hamburger from './hamburger.png';
import styles from './styles.css';

class Header extends Component {
    constructor(props) {
        super(props);
        this.state = {
            isMobile: false,
            isDesktop: true,
            navOpen: false,
        }
    }

    openMobNav = () => {
        this.setState({
            isMobile: true,
            isDesktop: false,
            navOpen: !this.state.navOpen,
        })
    }

    render() {
        const { navOpen } = this.state;
        return(
            <div className="container">
                <img 
                    className="hamburger"
                    alt="hamburger"
                    onClick={this.openMobNav}
                    src={hamburger} 
                    width={30} 
                    height={30} 
                    />
                { navOpen ? 
                    <ul className="resp-nav">{headerNames.sections.map((each, i) => 
                        <li className="resp-navbar-item" key={i}>{each.name}</li>)}
                    </ul>
                    : null 
                }
                {headerNames.sections.map((each, i) => {
                    return <a href={each.link} key={i} className="navbar-item">{each.name}</a>
                })}
            </div>
        )
    }
};

export default Header;
