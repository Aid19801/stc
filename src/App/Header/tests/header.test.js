import React from 'react';
import ReactDOM from 'react-dom';
import Header from '../header';
import { shallow } from 'enzyme';
import { headerNames } from '../header-names';


describe('Header Tests | ', () => {
    it('should render the Header component without crashing', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper.find('.container')).toHaveLength(1);
        expect(wrapper.find('.hamburger').hasClass('hamb-img')).toEqual(true);
    });
    it('should render the correct number of sections', () => {
        const wrapper = shallow(<Header />);
        const expectedNoOfHeaderNames = headerNames.sections.length;
        const renderedNoOfHeaderNames = wrapper.find('.navbar-item').length;
        expect(renderedNoOfHeaderNames).toEqual(expectedNoOfHeaderNames);
    });
    it('should render the Header with Desktop-True state as default', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper.state().isMobile).toEqual(false);
        expect(wrapper.state().isDesktop).toEqual(true);
        expect(wrapper.state().navOpen).toEqual(false);
    });
    it('should NOT render a navBar if on Desktop (navOpen state is false)', () => {
        const wrapper = shallow(<Header />);
        const expectedlyAbsentNav = '.resp-nav';
        expect(wrapper.state().isDesktop).toEqual(true);
        expect(wrapper.state().navOpen).toEqual(false);
        expect(wrapper.find(expectedlyAbsentNav).exists()).toBe(false); 
    });
    it('*should* render a NavBar when on Desktop because hamburger is clicked (navOpen state becomes true)', () => {
        const wrapper = shallow(<Header />);
        const hamburger = wrapper.find('.hamburger').at(0);
        hamburger.simulate('click');
        expect(wrapper.state().navOpen).toEqual(true);
        expect(wrapper.find('.navbar-item').exists()).toBe(true);
    });
    it('*should* render a NavBar then hide it, then show it again', () => {
        const wrapper = shallow(<Header />);
        const hamburger = wrapper.find('.hamburger').at(0);
        const isNavOpen = wrapper.state().navOpen;
        expect(isNavOpen).toBe(false);
        expect(wrapper.find('.resp-nav').exists()).toBe(false);

        hamburger.simulate('click');
        const isNavOpenAfter = wrapper.state().navOpen;
        expect(isNavOpenAfter).toBe(true);
        expect(wrapper.find('.resp-nav').exists()).toBe(true);

        hamburger.simulate('click');
        const isNavOpenAfterClickingAgain = wrapper.state().navOpen;
        expect(isNavOpenAfterClickingAgain).toBe(false);
        expect(wrapper.find('.resp-nav').exists()).toBe(false);
    });
    // it('should call openMobNav() when user clicks hamburger', () => {
    //     jest.mock('Header').openMobNav();
    //     const openMobNav = jest.fn();
    //     const wrapper = shallow(<Header />);
    //     const hamburger = wrapper.find('.hamb-img').at(0);
    //     hamburger.simulate('click')
    //     expect(openMobNav).toHaveBeenCalledTimes(1);
    // })
});
