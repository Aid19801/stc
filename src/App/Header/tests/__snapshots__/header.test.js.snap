// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests | ðŸŒ¹ should match the Header.js code snapshot 1`] = `
<div
  className="container"
>
  <img
    className="hamburger hamb-img"
    height={30}
    onClick={[Function]}
    src="hamburger.png"
    width={30}
  />
  <a
    className="navbar-item"
    href="#"
    key="0"
  >
    home
  </a>
  <a
    className="navbar-item"
    href="#"
    key="1"
  >
    about
  </a>
  <a
    className="navbar-item"
    href="#"
    key="2"
  >
    media
  </a>
  <a
    className="navbar-item"
    href="#"
    key="3"
  >
    gallery
  </a>
  <a
    className="navbar-item"
    href="#"
    key="4"
  >
    contact
  </a>
</div>
`;
